File: 2021/day_1.js
// https://adventofcode.com/2021/day/1

const INPUT = require('fs')
  .readFileSync(__dirname + '/day_1.txt', 'utf8')
  .trimEnd()
  .split('\n')
  .map(x => parseInt(x));

(function part_1() {
  let inc = 0;

  for (let i = 1; i < INPUT.length; i++) {
    if (INPUT[i] > INPUT[i - 1]) {
      inc++;
    }
  }

  console.log('part 1: %s', inc);
})();

(function part_2() {
  let inc = 0;

  for (let i = 3; i < INPUT.length; i++) {
    if (INPUT[i] > INPUT[i - 3]) {
      inc++;
    }
  }

  console.log('part 2: %s', inc);
})();

File: 2021/day_1.rs
// https://adventofcode.com/2021/day/1

fn main() {
    let input: Vec<usize> = include_str!("day_1.txt")
        .trim_end_matches('\n')
        .split('\n')
        .map(|x| x.parse::<usize>().unwrap())
        .collect();

    let part_1 = input
        .windows(2)
        .map(|x| if x[1] > x[0] { 1 } else { 0 })
        .sum::<usize>();

    println!("part 1: {:?}", part_1);

    let part_2 = input
        .windows(4)
        .map(|x| if x[3] > x[0] { 1 } else { 0 })
        .sum::<usize>();

    println!("part 2: {:?}", part_2);
}
